posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd

```diff
@@ -0,0 +1,252 @@
+
+# Introduction
+
+Date and time is collected in SDTM as character values using the extended [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format `"yyyy-dd-mmThh:mm:ss"`. This universal format allows missing parts date or time - e.g. the string`"2019-10"` represents a date where the day and the time are unknown. In contrast, ADaM timing variables like `ADTM` (Analysis Datetime) or `ADY` (Analysis Relative Day) are numeric variables, which can be derived only if the date or datetime is complete. 
+
+Most ADaM programmers have, at one point or another, encountered situations where missing dates, unexpected formats or confusing imputation functions rendered derivations of timing variables frustrating and time consuming. `{admiral}` aims to mitigate this (where possible!) by providing functions which automatically derive date/datetime variables for you, and fill in missing date or time parts according to well-defined imputation rules.
+
+In this article, we first examine the arsenal of functions provided by`{admiral}` to aid in datetime imputation and timing variable derivation. We then observe everything in action through a number of selected typical examples.
+
+# Date/Datetime Derivation and Imputation Functions
+
+`{admiral}` provides the following functions for date/datetime imputation:
+
+- `derive_vars_dt()`: Adds a date variable and a date imputation flag variable (optional) based on a --DTC variable and imputation rules.
+- `derive_vars_dtm()`: Adds a datetime variable, a date imputation flag variable, and a time imputation flag variable (both optional) based on a --DTC variable and imputation rules.
+- `impute_dtc_dtm()`: Returns a complete ISO 8601 datetime or `NA` based on a partial ISO 8601 datetime and imputation rules.
+- `impute_dtc_dt()`: Returns a complete ISO 8601 date (without time) or `NA` based on a partial ISO 8601 date(time) and imputation rules.
+- `convert_dtc_to_dt()`: Returns a date if the input ISO 8601 date is complete. Otherwise, `NA` is returned.
+- `convert_dtc_to_dtm()`: Returns a datetime if the input ISO 8601 date is complete (with missing time replaced by `"00:00:00"` as default). Otherwise, NA is returned.
+- `compute_dtf()`: Returns the date imputation flag.
+- `compute_tmf()`: Returns the time imputation flag.
+
+From the point of view of a typical ADaM programmer, the functions `impute_*`, `convert_*` and `compute_*` above can be viewed as utilities for treating dates and/or imputation within any custom code. In contrast, their `derive_*` find their use in directly deriving new timing variables and/or carrying out imputation at an ADaM dataset scale.
+
+For a detailed look at the Imputation rules applied by these `{admiral}` functions, please visit [this vignette](https://pharmaverse.github.io/admiral/cran-release/articles/imputation.html#imputation-rules) on the documentation website.
+
+# Action Examples
+
+## Creating an Imputed Datetime and Date Variable and Imputation Flag Variables
+
```

Comment:
What about starting off with a really easy example that uses the `impute_` functions and discuss the arguments.  Then segue into how the `derive_` are wrappers around the `impute` and `compute` functions?